crates/biome_js_analyze/src/assist/source/organize_imports.rs

```diff
@@ -1,143 +1,328 @@
+    /// // Detached comment
     ///
-    /// ```js
-    /// import "../styles/reset.css";
-    /// import "../styles/layout.css";
-    /// import { Grid } from "../components/Grid.jsx";
+    /// import * as C from "c";
     /// ```
     ///
-    /// Another common case is import polyfills or shim files, that needs to stay at the top file:
+    /// The import and export organizer ensures that chunks are separated from each other with a blank lines.
+    /// Only side-effect imports adjacent to a chunk are not separated by a blank line.
+    ///
+    /// Imports or exports of a chunk are sorted together.
+    /// Import and export sorting uses several layer of order.
+    ///
+    /// ## Import and export groups
+    ///
+    /// Imports or exports of a chunk are divided into groups.
+    /// By default, it exists a single group: the implicit group.
+    ///
+    /// You can define extra groups using the `groups` option.
+    /// The organizer provides predefined groups and allows you to create new groups based on a glob or a list of globs.
+    /// The predefined groups are:
     ///
-    /// ```js
-    /// import "../polyfills/array/flatMap";
-    /// import { functionThatUsesFlatMap } from "./utils.js";
+    /// - `:NODE:`: all imports and exports with a source starting by the protocol `node:` or that corresponds to a builtin Node,js module such as `path`
+    /// - `:BUN:`: all imports and exports with a source starting by the protocol `bun:`
+    ///
+    /// Given the following configuration and code, the import organizer propose to reorder the imports to place the Node.js imports first.
+    ///
+    /// ```json,full_options
+    /// {
+    ///     "assist": {
+    ///         "actions": {
+    ///             "source": {
+    ///                 "organizeImports": {
+    ///                     "level": "on",
+    ///                     "options": {
+    ///                         "groups": [
+    ///                             ":NODE:",
+    ///                             ":BUN:"
+    ///                         ]
+    ///                     }
+    ///                 }
+    ///             }
+    ///         }
+    ///     }
+    /// }
     /// ```
     ///
-    /// In these cases, Biome will sort all these three imports, and it might happen that the order will **break** your application.
+    /// ```js,expect_diagnostic,use_options
+    /// import { Database } from "bun:sqlite";
+    /// import lib from "@any/lib";
+    /// import test from "node:test";
+    /// import path from "path";
+    /// ```
     ///
-    /// To avoid this, create a "group" of imports. You create a "group" by adding a **new line** to separate the groups.
+    /// Note that, `import lib from "@any/lib"` doesn't match any explicitly set groups.
+    /// Imports and exports that don't match any user-provided groups, always match the implicit group that appears after all other groups.
+    /// Thus the import is placed at the end.
```

Comment:
Here are some suggestions for restructuring the example. Let's remember the pattern "Explain example, show example."

Remove `import lib from "@any/lib";` because it wasn't mentioned in the preview paragraph. 

This paragraph should end here, then move `## Import and export sources` after this one. It talks about our defaults, so let's keep the flow going, let's not jump from default/implicit to customisation and back again.

After that, create a new paragraph (`##`) about creating custom groups and let's put it at the very end. In the docs of all our rules, we first explain the rule's default behaviour and then explain the options.